stdlib/InteractiveUtils/src/codeview.jl

```diff
@@ -42,6 +42,17 @@ function warntype_type_printer(io::IO, @nospecialize(ty), used::Bool)
     nothing
 end
 
+# True if one can be pretty certain that the compiler handles this union well,
+# i.e. must be small with concrete types.
+function is_expected_union(u::Union)
+	Base.unionlen(u) < 4 || return false
+	ua, ub = u.a, u.b
+	if !Base.isdispatchelem(ua) || ua == Core.Box
+	    return false
+	end
+	ub isa Union ? is_expected_union(ub) : Base.isdispatchelem(ub)
```

Comment:
`ua` and `ub` handling must be symmetric
```suggestion
	for x in Base.uniontypes(u)
	    if !Base.isdispatchelem(ua) || ua == Core.Box
	        return false
	    end
        end
	return true
```